sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -1293,8 +1293,8 @@ class CastChecks extends ExprChecks {
   val calendarChecks: TypeSig = none
   val sparkCalendarSig: TypeSig = CALENDAR + STRING
 
-  val arrayChecks: TypeSig = ARRAY.nested(commonCudfTypes + DECIMAL_128_FULL + NULL +
-      ARRAY + BINARY + STRUCT + MAP) +
+  val arrayChecks: TypeSig = STRING + ARRAY.nested(commonCudfTypes + DECIMAL_128_FULL + NULL +
+      ARRAY + BINARY + STRUCT + MAP + STRING) +
```

Comment:
I think `commonCudfTypes` already covers STRING

Reply:
Thank you! I have removed the "STRING"